### DISCUSSION TITLE
2021.10.0b0

### DISCUSSION THREAD

**Discussion by Thomas55555** [Thomas55555/husqvarna_automower#104] [Announcements]
![downloads](https://img.shields.io/github/downloads/Thomas55555/husqvarna_automower/2021.10.0b0/total.svg)
@simontegelid added the websocket support. Thanks for that. Instead of polling we now have live updates of the mower. :) Maybe you have to refresh your API-key on the husqvarana website, if it's not working.

## What’s Changed

* Add websocket support (#102) @simontegelid

<hr /><em>This discussion was created from the release <a href='https://github.com/Thomas55555/husqvarna_automower/releases/tag/2021.10.0b0'>2021.10.0b0</a>.</em>

**Comment by simontegelid** [comment]
Woop woop! Thanks for assisting with the testing ☺️

**Reply by Thomas55555** [reply]
I could now also find out, the difference in the API-keys. If I debug the session with an old key, then in the self.token dictionary there is this: `'scope': 'iam:read'` and the new API-key shows `'scope': 'iam:read amc:api'`
Do you think we should implement an error in the integration to inform the users, that they have to refresh their API-key, if the scope is `iam:read` ?
Maybe not everyone is reading the release notes...
And can you push your other improvements. I also saw, that it wasn't possible to delete an entity before...
